### PR TITLE
[WB-7807] plot annotation + minor clean up

### PR DESCRIPTION
**Pull request by kptkin**
Description
-----------
What does the PR do?

WB-7807 will be resolved in the UI, but i took the opportunity to do a minor clean up of this logic.

Testing
-------
How was this PR tested?

This is just a minor refactor, the testing is to make sure the changes didn't break the existing tests.

### CODE REVIEW

File: wandb/sdk/internal/tb_watcher.py [hunk 1]
```diff
@@ -419,21 +418,19 @@ def _handle_event(self, event: "ProtoEvent", history: "TBHistory" = None) -> Non
...
-                panel_config = custom_chart_panel_config(item, key, key + "_table")
-                config = {"panel_type": "Vega2", "panel_config": panel_config}
-                chart_keys.append(key)
-                self._tbwatcher._interface.publish_config(
-                    val=config, key=("_wandb", "visualize", key)
+        chart_keys = set()
+        for k in row:
+            if isinstance(row[k], CustomChart):
+                chart_keys.add(k)
+                key = row[k].get_config_key(k)
```
Comment (KyleGoyette) [inline wandb/sdk/internal/tb_watcher.py:425]: won't this log the visualization at key `_wandb.visualize.key`?

**Review by KyleGoyette — COMMENTED** [review]
A couple questions, but otherwise I think it looks good

File: tests/wandb_test.py [hunk 2]
```diff
@@ -22,7 +21,7 @@ def test_log_step(live_mock_server, test_settings, parse_ctx):
 
 def test_log_custom_chart(live_mock_server, test_settings, parse_ctx):
     with wandb.init(settings=test_settings) as run:
-        custom_chart = create_custom_chart(
+        custom_chart = wandb.plot_table(
```
[the remark this answers is not in the document]
Comment (kptkin) [reply tests/wandb_test.py:24]: the old code did `plot_table` and called `create_custom_chart`... I could just have this restored and call it in wandb.plot_table

File: wandb/sdk/internal/tb_watcher.py (hunk 1 above)
Comment (kptkin) [reply wandb/sdk/internal/tb_watcher.py:425]: yeah that's what we had before didn't we?

File: tests/wandb_test.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (KyleGoyette) [reply tests/wandb_test.py:24]: Yeah you're right. Let's make the change you've made here

File: wandb/sdk/internal/tb_watcher.py (hunk 1 above)
Comment (KyleGoyette) [reply wandb/sdk/internal/tb_watcher.py:425]: Oh, oops the k we add to chart keys is NOT modified with the prefixes

**Review by KyleGoyette — APPROVED** [review]

**Pull request merged**